tapdb/sqlc/migrations/000020_asset_unique_key.up.sql

```diff
@@ -1,3 +1,43 @@
+-- First step is to make sure all duplicate assets are seen as spent, if their
+-- anchor transaction is spent. That way we can later also group on the spent
+-- column to remove duplicates.
+UPDATE assets
```

Comment:
I'm not sure if I grok this first query. How are we setting all duplicate assets here as being spent? What is it in the CTE that selects for duplicates exclusively?

Reply:
Because the duplication happened during address self-transfer all assets should be considered spent in the first place?

Reply:
No, the logic that updates the `spent` flag in the normal application flow will only do that for the "original" assets, not the duplicates. So in order for us to be able to properly de-duplicate the assets, we'll just make sure the `spent` flag is set correctly for all of them. Update the comment on the query as well.